packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -38,122 +93,602 @@ import '@spectrum-web-components/popover/sp-popover.js';
+            `;
+
+        return nothing;
+    }
+
+    public renderPicker(): TemplateResult | typeof nothing {
+        if (this.readonly) return nothing;
 
         return html`
-            <div class="picker">
-                ${this.renderStateIcons()}
-
-                <sp-picker-button
-                    class="picker-button"
-                    ?open=${this.open}
-                    ?quiet=${this.quiet || this.readonly}
-                    ?invalid=${this.invalid}
-                    ?disabled=${isDisabled}
-                    @click=${this.showPicker}
-                >
-                    <slot name="calendar-icon" slot="icon">
-                        <sp-icon-calendar></sp-icon-calendar>
-                    </slot>
-                </sp-picker-button>
-
-                <!-- TODO: Enable "receives-focus" when calendar is navigable via keyboard -->
-                <sp-overlay
-                    placement="bottom-start"
-                    receives-focus="false"
-                    .triggerElement=${this as HTMLElement}
-                    ?open=${this.open}
-                    @sp-closed=${this.hidePicker}
+            <sp-picker-button
+                ?open=${this.isCalendarOpen}
+                ?quiet=${this.quiet}
+                ?invalid=${this.invalid}
+                ?disabled=${this.disabled}
+                @click=${() => (this.isCalendarOpen = true)}
+            >
+                <slot name="calendar-icon" slot="icon">
+                    <sp-icon-calendar></sp-icon-calendar>
+                </slot>
+            </sp-picker-button>
+
+            <sp-overlay
+                .triggerElement=${this.input}
+                placement="top"
+                offset="0"
+                receives-focus="true"
+                ?open=${this.isCalendarOpen}
+                @sp-closed=${() => (this.isCalendarOpen = false)}
+            >
+                <sp-popover>
+                    <div class="popover-content">
+                        <sp-calendar
+                            .value=${this.value}
+                            .min=${this.min}
+                            .max=${this.max}
+                            @change=${this.handleChange}
+                        ></sp-calendar>
+                    </div>
+                </sp-popover>
+            </sp-overlay>
+        `;
+    }
+
+    private handleChange(event: Event): void {
+        event.stopPropagation();
+        this.isCalendarOpen = false;
+        const calendarValue = (event.target as Calendar).value as CalendarDate;
+
+        if (!this.includesTime) this.value = calendarValue;
+        else {
+            const hour = this.segments.hour?.value ?? 0;
+            const minute = this.segments.minute?.value ?? 0;
+            const second = this.segments.second?.value ?? 0;
+            const time = new Time(hour, minute, second);
+            this.value = toCalendarDateTime(calendarValue, time);
```

Comment:
I think here the value can also be a `ZonedDateTime` so a check needs to be added to know exactly into what type to convert the `CalendarDate`.

Reply:
Fixed in the next PR about precision updates.